tests/rfcml_bending/test_bending.py

```diff
@@ -0,0 +1,591 @@
+
+        _got = rms_forces
+        _expect = 5
+        # _perc_diff = 100 * np.abs(_expect - _got) / _expect
+        # _criteria = (0.1 / _expect) * 100  # Use sig-figs, but might be too tight
+
+        self.assertTrue(
+            _got <= _expect,
+            msg=f"Expected RMS of forces of {_expect:0.1f},"
+            "but got {_got:0.1f}, a % diff of {_perc_diff:0.1f}.",
+        )
+
+        _got = np.max(forces)
+        _expect = 11
+        _perc_diff = 100 * np.abs(_expect - _got) / _expect
+        # _criteria = (0.1 / _expect) * 100  # Use sig-figs, but might be too tight
+
+        self.assertTrue(
+            _got <= _expect,
+            msg=f"Expected max force of {_expect:0.1f},"
+            "but got {_got:0.1f}, a % diff of {_perc_diff:0.1f}.",
+        )
+
+    def test_calc_moments(self):
+        """Applies moments and verifies loads are balanced."""
+
+        # Load Steve's version of the force matrix
+        mat_file = TEST_FORCESPACE
+        bending = Bending(mat_file)
+        # force_space = load_force_space(mat_file)
+
+        # let's arbitrarily apply a force at 2 actuators
+        force1 = 2
+        force2 = 3
+        i1 = 25
+        i2 = 100
+        bending.force_space.data["FEA_act_ids"][i1]
+        bending.force_space.data["FEA_act_ids"][i2]
+
+        # calculate expectations
+        moment_x_exp = (
+            force1 * bending.force_space.data["x_act"][i1] + force2 * bending.force_space.data["x_act"][i2]
+        )
+        moment_y_exp = (
+            force1 * bending.force_space.data["y_act"][i1] + force2 * bending.force_space.data["y_act"][i2]
+        )
+
+        # Create a forces vector:
+        forces = np.zeros(len(bending.force_space.data["FEA_act_ids"]))
+        forces[i1] = force1
+        forces[i2] = force2
+
+        # Check that only a warning is thrown
+
+        with self.assertWarns(UserWarning):
+            mom_x, mom_y = bending.calc_moments(forces, assertion=True, warning_only=True)
+            self.assertAlmostEqual(moment_x_exp, mom_x)
+            self.assertAlmostEqual(moment_y_exp, mom_y)
+
+        # Check that an error is raised
+        with self.assertRaises(AssertionError):
+            mom_x, mom_y = bending.calc_moments(forces, assertion=True)
+
+        # Now load balance and check no errors are thrown.
+        # Note that the acceptable limit is unknown
+        # Assuming 0.1% of input.
+        f_balanced = bending.force_space.data["af2lc"] @ forces
+
+        limit = np.abs(0.001 * np.min([moment_x_exp, moment_y_exp]))
+        mom_x, mom_y = bending.calc_moments(f_balanced, limit=limit)
+
+    def test_fit_coeffs(self):
+        """Performs bending mode fit to itself to produce column vector of 1."""
+
+        mat_file = TEST_FORCESPACE
+        bending = Bending(mat_file)
+
+        modes = 5 # define number of modes you're fitting to
+
+        input_coeff = np.arange(1, modes + 1) ## define the input coefficients
```

Comment:
probably better to use random numbers, otherwise you will have all positive coefficients. It's probably fine, but better to use something more realistic

Reply:
Yes that! pull random coeffs but use the same seed for the generator.